sieve_of_eratosthenes.java

```diff
@@ -0,0 +1,34 @@
+
+// This program prints all the prime numbers from 2 to n
+
+
+import java.io.*;
+import java.util.*;
+class Solution
+{
+	public static void main(String args[])
+	{
+		Scanner sc=new Scanner(System.in);
+		int n=sc.nextInt();                   //n is the number upto which we need to find the prime numbers.
+	int arr[]=new int[n+1];              
+	for(int i=0;i<=n;i++)                 //initialize all the indexes with 1.
+	{
+		arr[i]=1;
+	}
+	for(int i=2;(i*i)<=n;i++)                //start from 2 as 2 is the first prime number.
+	{
+		if(arr[i]==1)                         
+		{
+		for(int j=i*2;j<=n;j+=i)               /* replace the value at all indexes that are multiples of j with 0 
+		{                                         ultimately we wiil be left with the numbers that are not multiples of any other number.*/
```

Comment:
you have commented this "curly brace" too, either use inline comment or move the brace to new line.